lib/hammer_cli_katello/content_view_purge.rb

```diff
@@ -0,0 +1,59 @@
+module HammerCLIKatello
+
+  class ContentViewPurge < HammerCLIKatello::Command
+    command_name "purge"
+    desc "Delete old versions of a content view"
+
+    option "--id", "ID", _("content view numeric identifier"), :required => true
+    option "--count", "COUNT", _("count of versions to keep"), :default => 3, :format => HammerCLI::Options::Normalizers::Number.new
+
+    def resource_content_views
+      HammerCLIForeman.foreman_resource(:content_views)
+    end
+
+    def resource_content_view_versions
+      HammerCLIForeman.foreman_resource(:content_view_versions)
+    end
+
+    def execute
+      if option_count < 0
+        print_message _("Invalid value for --count parameter: value must be greater than 0.")
+        HammerCLI::EX_USAGE
+      else
+        view = resource_content_views.call(:show, :id => option_id)
+
+        # Sort versions by the number (1.0, 2.0, ...)
+        all_versions = view['versions'].sort do |a, b|
+          a["version"].to_f <=> b["version"].to_f
```

Comment:
I'm not sure this would work. Suppose you have two versions 14.9 and 14.10; I think this comparison would tell us that 14.9 > 14.10 because 14.9 > 14.1 (to_f).

You could sort by the two columns `major` and `minor`:

```
all_versions = view['versions'].sort_by do |version|
  [version[:major], version[:minor]]
end
```

I believe this would work.